src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs

```diff
@@ -130,7 +133,15 @@ private void AddBodyParamsAndHeaders(IDictionary<string, string> additionalBodyP
             }
 
             _oAuth2Client.AddBodyParameter(OAuth2Parameter.Scope, scopes);
+
+#if SUPPORT_KERBEROS
```

Comment:
My main question is here - what is MSAL expected to do on non-Windows operating systems? We have 2 options: 

- throw a meaningful exception 
- do nothing

I generally prefer to throw a meaningful exception as it is more obvious. But if the flow can be made to work by providing a degraded experience, then maybe we can go down this route. How did you fix this problem for MSAL C++?

Reply:
For the AAD Kerberos Feature,
 - It can be supported for all OS and platforms.
 - One exception is for the Ticket Cache interface which is only supported in Windows OS platform now.

I'd make a change to conditionally compile the Windows Ticket Cache interface components using the (iOS || MAC || ANDROID) flag combination. 
- I'd changed the KerberosSupplementalTicketManager to throw exception when the user  call's Ticket Cache associated function in non-Windows OS platform.

With these changes, the whole AAD Kerberos feature  can be supported for all OS platforms.


In C++, there's a flag indicating the Windows OS platform, and all AAD Kerberos Features are only included for Windows target.